src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -99,8 +100,19 @@ internal static SettingBase Parse(XNode node, SettingsFile origin)
         internal static IEnumerable<T> ParseChildren<T>(XElement xElement, SettingsFile origin, bool canBeCleared) where T : SettingElement
         {
             var children = new List<T>();
-
-            var descendants = xElement.Elements().Select(d => Parse(d, origin)).OfType<T>().Distinct();
+            IEnumerable<T> descendants;
+            if (xElement.Name.LocalName.Equals("packageSourceMapping", StringComparison.OrdinalIgnoreCase))
```

Comment:
Better use constant defined at top instead of hard coded string `packageSourceMapping`, most likely we already have one.